### ISSUE TITLE
Action to enter copy mode and move cursor to selection

### ISSUE THREAD

**Issue by iloveitaly** [abhinav/tmux-fastcopy#155]
Hello, curious if you have any ideas about the best way to implement an action which:

1. Enters copy mode
2. Changes the cursor to be positioned as the selected text

**Comment by abhinav** [comment]
This is a great idea, @iloveitaly.

But once you have that, looking around `man tmux`, there's no easy way to do this unless I missed something.

`tmux copy-mode` doesn't let you specify where you want to place the cursor.
There is `tmux send-keys -X`, which lets you send copy-mode specific commands but that doesn't have a "go to this position" command.
There's a `goto-line <line>` command, but the line number is relative to the cursor's current position, but that only takes you up. It doesn't support going down from the cursor position.
There are `cursor-up` and `cursor-down`, but they operate like moving up/down in a text editor, moving the cursor to the end of each line, changing its horizontal position as you go, which mucks up the `x` position calculation.

From experimentation, assuming you have the match position (which you don't yet -- tmux-fastcopy will need to export that), you can get it to work if you do:

- Inspect the pane and get the `copy_cursor_x` and `copy_cursor_y` (these are measured from the top left).
- If the target position is above `copy_cursor_y`, `goto-line` to `$copy_cursor_y - $target_y + 1`
- Then `-N $((target_x - copy_cursor_x)) cursor-right` to get to the word start (cursor-left if the x position is to the left).
- `begin-selection` to start selection, `cursor-right` over the word, `end-selection`.

This won't work if the word is below the cursor.
Honestly, this feels unreliable enough that I wouldn't recommend doing it this way.

If you're not tied to selecting *that* instance of the word, if any instance of that word will suffice, a much better approach is this:

- send the pane into copy mode (`tmux copy-mode`)
- use `send-keys -X`:
    - `search-backward-text "$matchedText"` to move the cursor to the match
    - `begin-selection`, `-N $(length $matchedText - 1) cursor-right`, `end-selection` to select the match

This will be reliable and you can probably implement it today with what tmux-fastcopy already exports.
Although it may need to start exposing the pane ID of your original pane so that the above operations can be run on the pane you were working in, instead of tmux-fastcopy's pane. (tmux-fastcopy operates by creating a second pane with the text of your original pane and the UI to select text, and it swaps in its pane for yours, and swaps your back in when you pick something or cancel.)

So, in short, tmux-fastcopy probably needs to export a `FASTCOPY_TARGET_PANE` or an environment variable like that that your command implementation could use for the `search-backward-text` approach.

**Comment by abhinav** [comment]
@iloveitaly Looks like the search-backwards version above works quite well.
I was able to write a simple action that leveraged the above with the new `FASTCOPY_TARGET_PANE_ID` environment variable to select text.

#156 implements this new environment variable and also adds a new How-To entry explaining how to use it to select text instead of copying it.
I have to say, I like this enough that I'm making it my fastcopy-shift-action!

**Comment by iloveitaly** [comment]
@abhinav wow, this is awesome! Thanks so much for putting this together.
